### DISCUSSION TITLE
Recipe to set Select values dynamically?

### DISCUSSION THREAD

**Discussion by thomast8** [reflex-dev/reflex#3463] [Q&A] [answered]
I'm trying to implement a Select component whose values are not hardcoded but loaded dynamically.
To this end I have a `State.get_values()` function that generates these values and writes them to `State._values`.  As I couldn't get the Select to work as I would have expected (i.e. `rx.select(State._values)`) I tried to define a `read_values()` function that returns the `State._values` variable.

My frontend code is as follows:
```
def read_values() -> list[str]:
    return State._values

@rx.page(on_load=State.get_values)
def symbols() -> rx.Component:
    return rx.select(State.get_values)
```
while my backend is like this:
```
class State(rx.State):
    _values : list[str] = []

    def get_values(self):
        self._values = ["a","b"]
```

No matter what I try I can't get the Select to show the values that I have loaded into the State.  By debugging I see that `read_values()` is called before the page loads and that `State.get_values` is called after that, but then I can't get `read_values()` to be called again, so it makes sense that the component is displaying `[]`.

Am I approaching this the correct way or is there another way or dealing with this kind of problem?

EDIT:
I finally managed to get it to work by changing my code this way, even though I'm not sure what `@rx.cached_var` is doing:
frontend:
```
@rx.page(on_load=State.load_values)
def symbols() -> rx.Component:
    return rx.select(read_values())
```
while my backend is like this:
```
class State(rx.State):
    _values : list[str] = []

    @rx.cached_var
    def get_values(self) -> list[str]:
        values = self._values
        return values

    def load_values(self):
        self._values = ["a","b"]
```

**Comment by picklelo** [comment, marked as answer]
@thomastiotto take a look at our [select docs](https://reflex.dev/docs/library/forms/select) to see how to bind your select to state.

Your code can be simplified to be

```python
class State(rx.State):
    values : list[str] = []

    def load_values(self):
        self.values = ["a","b"]
```
and
```python
rx.page(on_load=State.load_values)
def symbols() -> rx.Component:
    return rx.select(State.values)
```
